Foundation/NSURLSession/http/HTTPURLProtocol.swift

```diff
@@ -449,13 +449,10 @@ extension _HTTPURLProtocol: _EasyHandleDelegate {
     fileprivate func notifyDelegate(aboutReceivedData data: Data) {
         guard let t = self.task else { fatalError("Cannot notify") }
         if case .taskDelegate(let delegate) = t.session.behaviour(for: self.task!),
-            let dataDelegate = delegate as? URLSessionDataDelegate,
-            let task = self.task as? URLSessionDataTask {
-            // Forward to the delegate:
-            guard let s = self.task?.session as? URLSession else { fatalError() }
-            s.delegateQueue.addOperation {
-                dataDelegate.urlSession(s, dataTask: task, didReceive: data)
-            }
+            let _ = delegate as? URLSessionDataDelegate,
+            let _ = self.task as? URLSessionDataTask {
+            // Forward to protocol client:
+            self.client?.urlProtocol(self, didLoad: data)
         } else if case .taskDelegate(let delegate) = t.session.behaviour(for: self.task!),
             let downloadDelegate = delegate as? URLSessionDownloadDelegate,
             let task = self.task as? URLSessionDownloadTask {
```

Comment:
We can't move this because the arguments `totalBytesWritten` and `totalBytesExpectedToWrite` takes values that are specific to HTTP now.  We may need to refactor this in future.